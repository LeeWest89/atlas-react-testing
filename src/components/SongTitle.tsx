interface SongProp {
  title?: string;
  artist?: string;
}

export default function SongTitle ({ title, artist }: SongProp) {
  return (
    <div>
      <h2 className="font-bold leading-8 mb-2 text-2xl text-deep-yellow">{title || "Unknown Title"}</h2>
      <p className="mb-4 text-wing-red">{artist || "Unknown Artist"}</p>
    </div>
  );
}