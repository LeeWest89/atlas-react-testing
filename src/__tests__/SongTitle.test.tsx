import { render } from "@testing-library/react";
import { expect, test } from "vitest";
import SongTitle from "../components/SongTitle";

//  Test SongTitle with title and artist
test("SongTitle with title and artist passed", () => {
  const { container } = render (
    <SongTitle title="Tidal Drift" artist="Echoes of the Sea" />
  );
  expect(container).toMatchSnapshot();
});

//  Test SongTitle with title and no artist
test("SongTitle with title and no artist passed", () => {
  const { container } = render (
    <SongTitle title="Tidal Drift" />
  );
  expect(container).toMatchSnapshot();
});

//  Test SongTitle with no title and an artist
test("SongTitle with title and no artist passed", () => {
  const { container } = render (
    <SongTitle artist="Echoes of the Sea" />
  );
  expect(container).toMatchSnapshot();
});

// Test SongTitle nothing passed
test("SongTitle with title and no artist passed", () => {
  const { container } = render (
    <SongTitle />
  );
  expect(container).toMatchSnapshot();
});
