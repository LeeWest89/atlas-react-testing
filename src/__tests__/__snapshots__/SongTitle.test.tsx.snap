// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SongTitle with title and artist passed 1`] = `
<div>
  <div>
    <h2
      class="font-bold leading-8 mb-2 text-2xl text-deep-yellow"
    >
      Tidal Drift
    </h2>
    <p
      class="mb-4 text-wing-red"
    >
      Echoes of the Sea
    </p>
  </div>
</div>
`;

exports[`SongTitle with title and no artist passed 1`] = `
<div>
  <div>
    <h2
      class="font-bold leading-8 mb-2 text-2xl text-deep-yellow"
    >
      Tidal Drift
    </h2>
    <p
      class="mb-4 text-wing-red"
    >
      Unknown Artist
    </p>
  </div>
</div>
`;

exports[`SongTitle with title and no artist passed 2`] = `
<div>
  <div>
    <h2
      class="font-bold leading-8 mb-2 text-2xl text-deep-yellow"
    >
      Unknown Title
    </h2>
    <p
      class="mb-4 text-wing-red"
    >
      Echoes of the Sea
    </p>
  </div>
</div>
`;

exports[`SongTitle with title and no artist passed 3`] = `
<div>
  <div>
    <h2
      class="font-bold leading-8 mb-2 text-2xl text-deep-yellow"
    >
      Unknown Title
    </h2>
    <p
      class="mb-4 text-wing-red"
    >
      Unknown Artist
    </p>
  </div>
</div>
`;
